applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -342,9 +338,126 @@ static void available_contexts_cb(struct bt_conn *conn, enum bt_audio_context sn
 	LOG_DBG("conn: %s, snk ctx %d src ctx %d\n", addr, snk_ctx, src_ctx);
 }
 
-static const struct bt_bap_unicast_client_cb unicast_client_cbs = {
+static int temp_cap_index_get(struct bt_conn *conn, uint8_t *index, uint8_t *num_caps)
+{
+	if (conn == NULL) {
+		LOG_ERR("No conn provided");
+		return -EINVAL;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(temp_cap); i++) {
+		if (temp_cap[i].conn == conn) {
+			*num_caps = temp_cap[i].num_caps;
+			*index = i;
+			return 0;
+		}
+	}
+
+	/* Connection not found in temp_cap, searching for empty slot */
+	for (int i = 0; i < ARRAY_SIZE(temp_cap); i++) {
+		if (temp_cap[i].conn == NULL) {
+			temp_cap[i].conn = conn;
+			*num_caps = temp_cap[i].num_caps;
+			*index = i;
+			return 0;
+		}
+	}
+
+	LOG_WRN("No more space in temp_cap");
+
+	return -ECANCELED;
+}
+
+static void pac_record_cb(struct bt_conn *conn, enum bt_audio_dir dir, const struct bt_codec *codec)
+{
+	int ret = 0;
+	uint8_t temp_cap_index;
+	uint8_t num_caps;
+
+	ret = temp_cap_index_get(conn, &temp_cap_index, &num_caps);
```

Comment:
Why would you need to ask for num_caps when it is already in temp_cap[i].num_caps? You only need the index, I believe

Reply:
True, but I thought it would make it clearer and cleaner.